server/src/transactions/infra/typeorm/repositories/transactions.repository.ts

```diff
@@ -33,4 +36,14 @@ export class TransactionsRepository {
 
     return this.transactionsRepository.save(transaction);
   }
+
+  async filterByStatus({
+    status,
+  }: FilterTransactionsByStatusDTO): Promise<Transaction[]> {
+    const transactions: Transaction[] = await this.transactionsRepository.find({
```

Comment:
Retorna isso direto:

`return this.transactionsRepository.find({ where: { status } });`